distributed/worker.py

```diff
@@ -2843,86 +2790,46 @@ def _notify_plugins(self, method_name, *args, **kwargs):
+            dep.key in self.data or dep.key in self.actors for dep in ts.dependencies
         )
 
-    def validate_key_ready(self, key):
-        assert key in pluck(1, self.ready)
-        assert key not in self.data
-        assert key not in self.executing
-        assert key not in self.waiting_for_data
+    def validate_task_ready(self, ts):
+        assert ts.key in pluck(1, self.ready)
+        assert ts.key not in self.data
+        assert ts.state != "executing"
+        assert not ts.waiting_for_data
         assert all(
-            dep in self.data or dep in self.actors for dep in self.dependencies[key]
+            dep.key in self.data or dep.key in self.actors for dep in ts.dependencies
         )
 
-    def validate_key_waiting(self, key):
-        assert key not in self.data
-        assert not all(dep in self.data for dep in self.dependencies[key])
+    def validate_task_waiting(self, ts):
+        assert ts.key not in self.data
+        assert ts.state == "waiting"
+        if ts.dependencies:
+            assert not all(dep.key in self.data for dep in ts.dependencies)
 
-    def validate_key(self, key):
+    def validate_task(self, ts):
         try:
-            state = self.task_state[key]
-            if state == "memory":
-                self.validate_key_memory(key)
-            elif state == "waiting":
-                self.validate_key_waiting(key)
-            elif state == "ready":
-                self.validate_key_ready(key)
-            elif state == "executing":
-                self.validate_key_executing(key)
-        except Exception as e:
-            logger.exception(e)
-            if LOG_PDB:
-                import pdb
-
-                pdb.set_trace()
-            raise
-
-    def validate_dep_waiting(self, dep):
-        assert dep not in self.data
-        assert dep in self.nbytes
-        assert self.dependents[dep]
-        assert not any(key in self.ready for key in self.dependents[dep])
-
-    def validate_dep_flight(self, dep):
-        assert dep not in self.data
-        assert dep in self.nbytes
-        assert not any(key in self.ready for key in self.dependents[dep])
-        peer = self.in_flight_tasks[dep]
-        assert dep in self.in_flight_workers[peer]
-
-    def validate_dep_memory(self, dep):
-        assert dep in self.data or dep in self.actors
-        assert dep in self.nbytes
-        assert dep in self.types
-        if dep in self.task_state:
-            assert self.task_state[dep] == "memory"
-
-    def validate_dep(self, dep):
-        try:
-            state = self.dep_state[dep]
-            if state == "waiting":
-                self.validate_dep_waiting(dep)
-            elif state == "flight":
-                self.validate_dep_flight(dep)
-            elif state == "memory":
-                self.validate_dep_memory(dep)
-            else:
-                raise ValueError("Unknown dependent state", state)
```

Comment:
Just checking, can you verify that these checks were moved to the other validate methods?

Most of our testing confidence comes from running these methods while operating Dask in different situations.  It's very important to us that we not lose validation here.

Reply:
Thanks for the call out.  Deps are gone now, but all of these checks (except for `flight`) are covered in `validate_task_state`.  I'm adding an additional check for `flight` now, which should then cover all of these bases.